### ISSUE TITLE
Build fails with custom test regexp

### ISSUE THREAD

**Issue by dmkelly** [FatehAK/vite-plugin-image-optimizer#9]
### Describe the Bug

When running a build with a custom `test` regexp option, I get the following error:

```
[vite-plugin-image-optimizer] options.test.test is not a function
error during build:
TypeError: options.test.test is not a function
    at ./node_modules/vite-plugin-image-optimizer/dist/index.js:216:30
    at Array.forEach (<anonymous>)
    at Object.closeBundle (./node_modules/vite-plugin-image-optimizer/dist/index.js:215:20)
```

This appears to be related to the `deepClone` function that is called as part of merging the provided options with the default options. This function checks for whether a value is an object, but doesn't account for the value being a regexp. The `test` value ends up being replaced with an empty object. The following additional check for if a value is a regexp seems to fix builds:

```javascript
function deepClone(src) {
  const isRegex = Object.prototype.toString.call(src) === "[object RegExp]";
  if (typeof src !== "object" || src === null || isRegex)
    return src;
  const target = Array.isArray(src) ? [] : {};
  for (const key in src) {
    const value = src[key];
    target[key] = deepClone(value);
  }
  return target;
}
```

### Steps to reproduce

```javascript
ViteImageOptimizer({
  test: /\.(jpe?g|png|gif|tiff|webp|avif)$/i,
}),
```

### System Info

```shell
System:
    OS: macOS 13.2
    CPU: (10) arm64 Apple M2 Pro
    Memory: 82.55 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.13.0 - ~/.nvm/versions/node/v18.13.0/bin/node
    npm: 8.19.3 - ~/.nvm/versions/node/v18.13.0/bin/npm
  Browsers:
    Chrome: 109.0.5414.119
    Safari: 16.3
  npmPackages:
    @vitejs/plugin-react: 3.1.0 => 3.1.0
    vite: 4.1.1 => 4.1.1
    vite-plugin-environment: 1.1.3 => 1.1.3
    vite-plugin-fonts: 0.7.0 => 0.7.0
    vite-plugin-image-optimizer: 1.0.7 => 1.0.7
    vite-plugin-minify: 1.5.2 => 1.5.2
    vite-plugin-preload: 0.2.0 => 0.2.0
    vite-plugin-svgr: 2.4.0 => 2.4.0
    vite-tsconfig-paths: 4.0.5 => 4.0.5
```

### Used Package Manager

npm

### Validations

- [X] Read the [docs](https://github.com/FatehAK/vite-plugin-image-optimizer#readme).
- [X] Check that there isn't [already an issue](https://github.com/FatehAK/vite-plugin-image-optimizer/issues) that reports the same bug to avoid creating a duplicate.

**Comment by FatehAK** [comment]
Thanks for finding this issue! Will review and merge your PR shortly.

**Comment by FatehAK** [comment]
Just published v1.0.8 with the fix. Thanks again!

**Comment by dmkelly** [comment]
Thank you!
